feature_engine/encoding/similarity_encoder.py

```diff
@@ -0,0 +1,298 @@
+    (transform).
+
+    Parameters
+    ----------
+    top_categories: int, default=None
+        If None, dummy variables will be created for each unique category of the
+        variable. Alternatively, we can indicate in the number of most frequent
+        categories to encode. In this case, similarity variables will be created
+        only for those popular categories and the rest will be ignored.
+
+    handle_missing : str, default='impute'
+        Action to perform when NaN is seen.
+            'error' - raise an error;
+            'impute' - impute NaN with an empty string;
+            'ignore' - ignore NaN and leave them in resulting columns.
+
+    {variables}
+
+    {ignore_format}
+
+    Attributes
+    ----------
+    encoder_dict_:
+        Dictionary with the categories for which dummy variables will be created.
+
+    {variables_}
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    fit:
+        Learn the unique categories per variable
+
+    {fit_transform}
+
+    transform:
+        Replace the categorical variables by the binary variables.
+
+    Notes
+    -----
+    This encoder will encode new categories by measuring string similarity between
+    seen and unseen categories.
+
+    No preprocessing is applied, so it's on user to prepare string categorical
+    variables for this transformer.
+
+    The original categorical variables are removed from the returned dataset when we
+    apply the transform() method. In their place, the binary variables are returned.
+
+    See Also
+    --------
+    feature_engine.encoding.OneHotEncoder
+    dirty_cat.SimilarityEncoder
+
+    References
+    ----------
+    .. [1] Cerda P, Varoquaux G, Kégl B. "Similarity encoding for learning with dirty
+       categorical variables". Machine Learning, Springer Verlag, 2018.
+    .. [2] Cerda P, Varoquaux G. "Encoding high-cardinality string categorical
+       variables". IEEE Transactions on Knowledge & Data Engineering, 2020.
+    """
+
+    def __init__(
+        self,
+        top_categories: Union[None, int] = None,
+        handle_missing: str = 'impute',
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        ignore_format: bool = False,
+    ):
+        if top_categories and not isinstance(top_categories, int):
+            raise ValueError(
+                "top_categories takes only integer numbers, 1, 2, 3, etc."
+            )
+        if handle_missing not in ('error', 'impute', 'ignore'):
+            raise ValueError(
+                "handle_missing should be one of 'error', 'impute' or 'ignore'"
+            )
+        super().__init__(variables, ignore_format)
+        self.top_categories = top_categories
+        self.handle_missing = handle_missing
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        Learns the unique categories per variable. If top_categories is indicated,
```

Comment:
what happens if the user selects 5 top categories, and then categories 5 and 6 are equally popular? 

At the moment, our OneHotEncoder would select category 5th and ignore 6th, but we have an issue to try and fix this behaviour so that it selects both. Should we incorporate this here as well?

Reply:
I don't know about this issue, can you send me a link to it?

Reply:
#426

Reply:
I don't think this is an issue. The one category that will be taken among tied will be first in alphabetical order, which is consistent and this is all that matters.

When we define top_k we want to get k features out, not k-1 or k+1 or k+n. This parameter exist to define number of features to be outputted. If you implement this fix of adding all tied categories you will cause more questions and problems to the end user.